eth/backend.go

```diff
@@ -237,14 +238,17 @@ func New(stack *node.Node, config *ethconfig.Config) (*Ethereum, error) {
 	}
 	eth.APIBackend.gpo = gasprice.NewOracle(eth.APIBackend, gpoParams)
 
-	eth.ethDialCandidates, err = setupDiscovery(eth.config.EthDiscoveryURLs)
+	// Setup DNS discovery iterators.
+	dnsclient := dnsdisc.NewClient(dnsdisc.Config{})
+	eth.ethDialCandidates, err = dnsclient.NewIterator(eth.config.EthDiscoveryURLs...)
```

Comment:
Out of curiosity, is this safe if `eth.config.EthDiscoveryURLs` == `[]`? Previously we had a special clause that returned `nil, nil` for that, but here we're instantiating a random iterator even for no-urls. Just want a confirm that we won't spin loop shomewhere.

Reply:
Yes, it should be safe.

Reply:
But let me try it anyway.

Reply:
I've tested it, and there is no problem with empty list. The iterator is basically closed immediately.